src/task.c

```diff
@@ -707,17 +657,48 @@ task_group_perform_one(task_group *group)
 platform_status
 task_perform_one(task_system *ts)
 {
-   platform_assert(!ts->use_bg_threads);
    platform_status rc = STATUS_OK;
    for (task_type type = TASK_TYPE_FIRST; type != NUM_TASK_TYPES; type++) {
-      rc = task_group_perform_one(&ts->group[type]);
+      rc = task_group_perform_one(&ts->group[type], 0);
+      if (STATUS_IS_NE(rc, STATUS_TIMEDOUT)) {
+         return rc;
+      }
+   }
+   return rc;
+}
+
+platform_status
+task_perform_one_if_needed(task_system *ts)
+{
+   platform_status rc = STATUS_OK;
+   for (task_type type = TASK_TYPE_FIRST; type != NUM_TASK_TYPES; type++) {
+      rc = task_group_perform_one(&ts->group[type],
+                                  ts->group[type].bg.num_threads);
```

Comment:
Comment explaining what this policy means

Reply:
Agreed; some comments will be helpful. I might put them in the function's prolog.

I do have a question whether this heuristic will address some of the issues we've seen when running large-inserts stress test. For example, under issue #474 (and perhaps, similar configuration problem is also the RCA for the issue #310), we have a situation where trunk # bundles is inadequate.

Does it likely that such problems may be resolved with foreground threads picking up some work based on configured # of bg-threads?

I guess this policy may need to be tweaked a bit to get more such stress tests running stably.

I am wondering (just asking aloud ...) whether we should make fg-threads be more aggressive by changing the `min_backlog` arg to `ts->group[type].bg.num_threads / 2`? 

I know it's all a bit arbitrary. Just wanted to hear your thoughts on how we could best refine this heuristic.

Reply:
I agree the policy is a bit arbitrary.  However, I don't think this is the place to solve liveness bugs in trunk.c.